packages/workspace/src/generators/convert-to-buildable/convert-to-buildable.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  NxJsonConfiguration,
+  readJsonFile,
+  readProjectConfiguration,
+  formatFiles,
+  Tree,
+  writeJson,
+  convertNxGenerator,
+  updateProjectConfiguration,
+} from '@nrwl/devkit';
+
+import { Schema } from './schema';
+import { join } from 'path';
+
+function getExecutor(type: Schema['type']): string {
+  switch (type) {
+    case 'detox':
+    case 'js':
+    case 'web':
+      return '@nrwl/js:tsc';
+    case 'next':
+    case 'react':
+      return '@nrwl/web:rollup';
+    case 'node':
+    case 'nest':
+      return '@nrwl/node:package';
+  }
+}
+
+export async function convertToBuildable(host: Tree, schema: Schema) {
+  const configuration = readProjectConfiguration(host, schema.project);
+
+  if (configuration.projectType !== 'library') {
+    throw new Error(`${schema.project} is not a library.`);
+  }
+
+  const nxJson = readJsonFile<NxJsonConfiguration>('nx.json');
+
+  if (configuration.targets['build'] != null) {
+    throw new Error(`${schema.project} is already buildable.`);
+  }
```

Comment:
This check makes sense if we keep going with only single-project command. So this isn't necessarily a change request, but rather a discussion point. Should we change this generator to take --all similar to how it works with the convert-to-nx-project generator? Should it be able to take a list of projects instead of only 1? 

Since buildable libs can only depend on other buildable libs, I feel like the most common DX will be converting **_all_** libs that are in an apps dependency tree to buildable at the same time.

Reply:
I would say yes we do want to be able to support that (and it's something I thought about while building this). Just a couple of things to consider:

1. We would need to filter projects to only supported framework/type
2. This Pr doesn't yet support Angular libs but that is handled by #1